akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -151,6 +155,8 @@ private[akka] object FanIn {
 
     def isCancelled(input: Int): Boolean = cancelled(input)
 
+    def isAllCompleted(): Boolean = (0 to inputCount - 1).map(completed).forall(_ == true)
```

Comment:
If it is easy to add, then please do it. At least it will indicate if something has changed.